Description: Fix unit tests that fail on losetup
Author: Lorenzo "Palinuro" Faletra <palinuro@parrotsec.org>
Last-Update: 2022-08-07

--- grub2-2.06.orig/tests/ext234_test.in
+++ grub2-2.06/tests/ext234_test.in
@@ -25,6 +25,12 @@ if ! which mkfs.ext4 >/dev/null 2>&1; th
    exit 77
 fi
 
+losetup -f
+if ! [ $? -eq 0 ]; then
+   echo "build environment does not support loop devices, cannot test ext234"
+   exit 77
+fi
+
 "@builddir@/grub-fs-tester" ext2_old
 "@builddir@/grub-fs-tester" ext2
 "@builddir@/grub-fs-tester" ext3
--- grub2-2.06.orig/tests/fat_test.in
+++ grub2-2.06/tests/fat_test.in
@@ -15,6 +15,12 @@ if ! which mkfs.vfat >/dev/null 2>&1; th
    exit 77
 fi
 
+losetup -f
+if ! [ $? -eq 0 ]; then
+   echo "build environment does not support loop devices, cannot test ext234"
+   exit 77
+fi
+
 "@builddir@/grub-fs-tester" vfat16a
 "@builddir@/grub-fs-tester" vfat12a
 "@builddir@/grub-fs-tester" vfat12
